shared/js/content-scripts/click-to-load.js

```diff
@@ -462,6 +462,58 @@
             return frame
         }
 
+        /*
+         * Tweaks an embedded YouTube video element ready for reinsertion to the
+         * page.
+         *
+         * @param {Element} videoElement
+         * @returns {Function?} onError
+         *   Function to be called if the video fails to load.
+         */
+        async adjustYouTubeVideoElement (videoElement) {
+            let onError = null
+
+            if (!videoElement.src) {
+                return onError
+            }
+            const url = new URL(videoElement.src)
+            const { hostname: originalHostname } = url
+
+            // Upgrade video to YouTube's "privacy enhanced" mode, but fall back
+            // to standard mode if the video fails to load.
+            // Note:
+            //  1. Changing the iframe's host like this won't cause a CSP
+            //     violation on Chrome, see https://crbug.com/1271196.
+            //  2. The onError event doesn't fire for blocked iframes on Chrome.
```

Comment:
How are you handling this case? Should we check if the frame has loaded?

Reply:
The CSP violation won't happen on Chrome, so it doesn't matter that the `onError` event won't fire since we're only  using the `onError` event to fix the CSP violation.